persistent/persistence.py

```diff
@@ -95,12 +98,18 @@ def _get_oid(self):
     def _set_oid(self, value):
         if value == self.__oid:
             return
-        if value is not None:
-            if not isinstance(value, OID_TYPE):
-                raise ValueError('Invalid OID type: %s' % value)
-        if self.__jar is not None and self.__oid is not None:
-            raise ValueError('Already assigned an OID by our jar')
-        self.__oid = value
+        # The C implementation allows *any* value to be
+        # used as the _p_oid.
```

Comment:
On Tuesday, April 28, 2015 07:02:48 AM Jim Fulton wrote:
> When I first created ZODB, the type of OIDs was left undefined. Over the
> years, there have been implementations that used (or experimented with)
> various types, including long strings (APE) and tuples.  Also, over the
> years, oids have settled in 64-bit integers encoded as big-endian strings.
> 
> In the future (or more likely in a successor, if it ever happened :) ), I
> expect to switch to UUIDs. <shrug>

Please do not require a specific object type for oids. I use the persistent 
package with mongopersist and pjpersist which have objects as OIDs. For 
mongopersist its pymongo..dbref.DBRef and for pjpersist it is a package-
specific type similar to DBRef.

Note that I am not using the ZODB package at all, just persistent and 
transaction.

Regards.
Stephan
-- 
Stephan Richter
Entrepreneur & Geek